spec/jobs/availability_status_update_job_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'rails_helper'
+
+RSpec.describe AvailabilityStatusUpdateJob, :type => :job do
+  let(:now)        { Time.new(2020).utc }
+  let(:payload)    { {"resource_type" => resource_type, "resource_id" => resource_id, "status" => status, "error" => reason}.to_json }
+  let(:headers)    { {"SECRET_HEADER" => "PASSWORD"} }
+  let(:reason)     { "host unreachable" }
+  let(:status)     { "unavailable" }
+
+  subject { AvailabilityStatusUpdateJob }
+
+  context "when body contains valid resource_type and id" do
+    let(:endpoint) do
+      create(:endpoint, :role => "first", :default => true)
+    end
+    let(:resource_type) { "endpoint" }
+    let(:resource_id)   { endpoint.id.to_s }
+
+    before { allow(Time).to receive(:now).and_return(now) }
```

Comment:
using Timecop will look better

Reply:
First time using it, but fixed it to use Timecop.